kucher/view/main_window/telega_control_widget/task_specific_status_widget/fault_status_widget.py

```diff
@@ -83,12 +89,38 @@ def on_general_status_update(self, timestamp: float, s: GeneralStatusView):
         self._error_code_hex.setText(f'0x{tssr.failed_task_exit_code:02X}')
         self._error_code_bin.setText(f'0b{tssr.failed_task_exit_code:08b}')
 
-        self._error_description_display.setText('(elaboration not available)')
+        file_name = get_absolute_path('view', 'main_window', 'telega_control_widget', 'task_specific_status_widget',
+                                      f'error_codes.yml', check_existence=True)
+        with open(file_name, 'r') as f:
+            error_codes = yaml.safe_load(f)
+
+        failed_task_name = str(tssr.failed_task_id).split('.')[-1]
+
+        error = error_codes[failed_task_name].get(tssr.failed_task_exit_code, 'unknown error')
+        error_description = error.get('description', 'unknown error') if type(error) is dict else error
+        error_comment = error.get('comment', '') if type(error) is dict else ''
```

Comment:
```suggestion
        error_comment = error.get('comment', '') if isinstance(error, dict) else ''
```

Applies to the line above as well. Your version breaks for subclasses.